pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -74,6 +74,31 @@ func (v *ControlPlaneMutator) Handle(ctx context.Context, req admission.Request)
 		}
 	}
 
+	newOpenShiftRoute := mutator.NewOpenShiftRoute()
+	if newOpenShiftRoute == nil {
+		oldVersion, err := versions.ParseVersion(mutator.OldVersion())
+
+		if err == nil {
+			oldOpenShiftRoute := mutator.OldOpenShiftRoute()
+
+			if oldOpenShiftRoute == nil {
+				switch req.AdmissionRequest.Operation {
+				case admissionv1beta1.Create:
+					mutator.SetOpenShiftRoute(false)
+				case admissionv1beta1.Update:
+					if oldVersion.LessThan(versions.V2_5) {
+						mutator.SetOpenShiftRoute(true)
+					} else {
+						mutator.SetOpenShiftRoute(false)
+					}
+				}
+			} else {
+				mutator.SetOpenShiftRoute(*oldOpenShiftRoute)
```

Comment:
If openshiftRoute is enabled in the old object, why are we applying a patch to enable it?

Reply:
Because pre 2.5, IOR is enabled by default. If we update the 2.5 charts and the base chart, an user, who is using IOR, could have the routes disabled as described by Rob's [comment](https://github.com/maistra/istio-operator/pull/1578#issuecomment-1892415559).  Using the mutation hook, we can check if it is an upgrade from lower version. If it is, we need to make sure IOR should continue to work.